ballerina/tests/to_xml_test.bal

```diff
@@ -0,0 +1,852 @@
+    xml result = check toXml(data);
+    test:assertEquals(result.toString(), expected, msg = "testComplexRecordToXml result incorrect");
+}
+
+@Namespace {
+    uri: "example.com"
+}
+type Purchesed_Bill record {
+    Purchesed_Items PurchesedItems;
+    Purchesed_Address Address;
+    @Attribute
+    string 'xmlns\:ns?;
+    @Attribute
+    string attr?;
+    @Attribute
+    string ns\:attr?;
+};
+
+type Purchesed_Items record {
+    Purchesed_Purchase[] PLine;
+};
+
+type Purchesed_Purchase record {
+    string|Purchesed_ItemCode ItemCode;
+    int Count;
+};
+
+type Purchesed_ItemCode record {
+    @Attribute
+    string discount;
+    string \#content?;
+};
+
+@Namespace {
+    uri: ""
+}
+type Purchesed_Address record {
+    string StreetAddress;
+    string City;
+    int Zip;
+    string Country;
+};
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testRecordWithNamaspaceAnnotationToXml() returns error? {
+    Purchesed_Bill input = {
+        PurchesedItems: {
+                PLine: [
+                    {ItemCode: "223345", Count: 10},
+                    {ItemCode: "223300", Count: 7},
+                    {
+                        ItemCode: {discount: "22%", \#content: "200777"},
+                        Count: 7
+                    }
+            ]
+        },
+        Address: {
+            StreetAddress: "20, Palm grove, Colombo 3",
+            City: "Colombo",
+            Zip: 300,
+            Country: "LK"
+        },
+        'xmlns\:ns: "ns.com",
+        attr: "attr-val",
+        ns\:attr: "ns-attr-val"
+    };
+    string expected =
+        "<Purchesed_Bill xmlns=\"example.com\" xmlns:ns=\"ns.com\" attr=\"attr-val\" ns:attr=\"ns-attr-val\">" +
+            "<PurchesedItems>" +
+                "<PLine>" +
+                    "<ItemCode>223345</ItemCode>" +
+                    "<Count>10</Count>" +
+                "</PLine>" +
+                "<PLine>" +
+                    "<ItemCode>223300</ItemCode>" +
+                    "<Count>7</Count>" +
+                "</PLine>" +
+                "<PLine>" +
+                    "<ItemCode discount=\"22%\">200777</ItemCode>" +
+                    "<Count>7</Count>" +
+                "</PLine>" +
+            "</PurchesedItems>" +
+            "<Address>" +
+                "<StreetAddress>20, Palm grove, Colombo 3</StreetAddress>" +
+                "<City>Colombo</City>" +
+                "<Zip>300</Zip>" +
+                "<Country>LK</Country>" +
+            "</Address>" +
+        "</Purchesed_Bill>";
+    xml result = check toXml(input);
+    test:assertEquals(result.toString(), expected, msg = "testComplexRecordToXml result incorrect");
+}
+
+@Namespace {
+    uri: "example.com"
+}
+type Purchesed_Bill1 record {
+    Purchesed_Items1 PurchesedItems;
+    @Attribute
+    string 'xmlns\:ns?;
+    @Attribute
+    string attr?;
+    @Attribute
+    string ns\:attr?;
+};
+
+@Namespace {
+    prefix: "ns0",
+    uri: "example.com"
+}
```

Comment:
isn't this namespace applied to the `PurchesedItems` element?

Reply:
No. Here, `Purchesed_Items1 ` is a subtype. So, it will not be considered in the conversion. If the user adds a namespace annotation to this, this will create a namespace inside that.
https://github.com/ballerina-platform/module-ballerina-xmldata/pull/431/files#diff-e33ed6da109b4de2fffd39b6d4d3d05cd574e38efbbfea5c8e5f014f41c38243R760

Reply:
Fixed in. [e9ed663](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/431/commits/e9ed663e63027a80cd23907bc746b35aae654804)